### PR TITLE
[RLlib] PG policy subclassing conversion

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

To include PG in the new policy subclassing scheme.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/pg/pg_tf_policy.py [hunk 1]
```diff
@@ -2,81 +2,143 @@
...
+            self.maybe_initialize_optimizer_and_loss()
+
+        @override(base)
+        def loss(
+            self,
+            model: ModelV2,
+            dist_class: Type[ActionDistribution],
+            train_batch: SampleBatch,
+        ) -> Union[TensorType, List[TensorType]]:
+            """The basic policy gradients loss function.
```
Comment (sven1977) [inline rllib/algorithms/pg/pg_tf_policy.py:80]: Can we add the math here?

Something like:
L = E[-log(pi(s,a)) * R]

File: rllib/policy/dynamic_tf_policy_v2.py [hunk 2]
```diff
@@ -1045,4 +1045,4 @@ def gradients(self, optimizer, loss):
             else:
                 return self.compute_gradients_fn(optimizers[0], losses[0])
         else:
-            return super().gradients(self, optimizers, losses)
+            return super().gradients(optimizers, losses)
```
Comment (sven1977) [inline rllib/policy/dynamic_tf_policy_v2.py:1048]: Nice catch!

**Review by sven1977 — APPROVED** [review]
Looks good, thanks @ArturNiederfahrenhorst !
Just a few nits, then we can merge.

**Review by gjoliver — APPROVED** [review]
appreciate

**Review by avnishn — APPROVED** [review]

**Review by sven1977 — APPROVED** [review]
Looks great. Thanks for the PR @ArturNiederfahrenhorst  and the smaller cleanups (e.g. `np.bool`)! :)

**Pull request merged**